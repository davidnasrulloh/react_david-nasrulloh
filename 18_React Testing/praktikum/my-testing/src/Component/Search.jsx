import React, {useState} from 'react';
import axios from "axios";

const URL = 'http://hn.algolia.com/api/v1/search';

export default function Search() {
  const [stories, setStories] = useState([]);
  const [error, setError] = useState(null);
  const [value, setValue] = useState("")

  async function handleFetch(event) {
    let result;

    try {
      result = await axios.get(`${URL}?query=${value}`);
      setStories(result.data.hits);
      console.log(value, "value", result.data.hits[0])
    } catch (error) {
      setError(error);
    }
  }

  return (
    <div>
      <h2>Cari Cerita</h2>
      <input onChange={e=> setValue(e.target.value)} type="text" placeholder="Tulis Cerita" />
      <button type="button" onClick={handleFetch}>
        Cari Cerita
      </button> <br/>

      {error && <span>Ada yang error ...</span>}

      {stories.length > 0 && "Daftar Cerita"}
      <ul>
        {stories.map((story) => (
          <li key={story.objectID}>
            {story.title} <button href={story.url}>Baca Selengkapnya</button>
          </li>
        ))}
      </ul>
    </div>
  );
}
