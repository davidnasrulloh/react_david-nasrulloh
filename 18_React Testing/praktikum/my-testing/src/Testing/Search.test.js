import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import Search from "../Component/Search";

describe("Search", () => {
    it('passing test', () => {
        expect(true).toBeTruthy();
    })

    it('failing test', () => {
    expect(false).toBeFalsy();
    })

    test("Render Search Component", () => {
        const tree = renderer
            .create(<Search/>)
            .toJSON()

        expect(tree).toMatchSnapshot();
    });

    test("Header Content", () => {
        render(<Search/>);
        expect(screen.getByText("Cari Cerita")).toBeInTheDocument();
        screen.debug();
    });

    test("Call the onChange callback handler", () => {
        const onChange = jest.fn();
        render(
            <Search value="" onChange={onChange}>
                Ini isi Search
            </Search>
        )
        fireEvent.change(screen.getByRole("textbox"),{
            target: {value: 'Java Script'}
        })
        expect(onChange).toHaveBeenCalledTimes(1);
    });
});
