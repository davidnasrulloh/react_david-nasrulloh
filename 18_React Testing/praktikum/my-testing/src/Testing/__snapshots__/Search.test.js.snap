// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Render Search Component 1`] = `
<div>
  <h2>
    Cari Cerita
  </h2>
  <input
    onChange={[Function]}
    placeholder="Tulis Cerita"
    type="text"
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Cari Cerita
  </button>
   
  <br />
  <ul />
</div>
`;
